service/common/src/main/java/org/apache/polaris/service/events/AwsCloudWatchEventListener.java

```diff
@@ -0,0 +1,228 @@
+import software.amazon.awssdk.services.cloudwatchlogs.model.InputLogEvent;
+import software.amazon.awssdk.services.cloudwatchlogs.model.InvalidSequenceTokenException;
+import software.amazon.awssdk.services.cloudwatchlogs.model.LogStream;
+import software.amazon.awssdk.services.cloudwatchlogs.model.PutLogEventsRequest;
+import software.amazon.awssdk.services.cloudwatchlogs.model.PutLogEventsResponse;
+import software.amazon.awssdk.services.cloudwatchlogs.model.ResourceAlreadyExistsException;
+
+@ApplicationScoped
+@Identifier("aws-cloudwatch")
+public class AwsCloudWatchEventListener extends PolarisEventListener {
+  private static final Logger LOGGER = LoggerFactory.getLogger(AwsCloudWatchEventListener.class);
+  private final ObjectMapper objectMapper = new ObjectMapper();
+  private static final int MAX_BATCH_SIZE = 10_000;
+  private static final int MAX_WAIT_MS = 5000;
+
+  private final BlockingQueue<EventAndTimestamp> queue = new LinkedBlockingQueue<>();
+  private CloudWatchLogsClient client;
+  private volatile String sequenceToken;
+
+  private volatile boolean running = true;
+
+  ExecutorService executorService;
+
+  private Future<?> backgroundTask;
+
+  private final String logGroup;
+  private final String logStream;
+  private final Region region;
+
+  @Inject
+  public AwsCloudWatchEventListener(
+      EventListenerConfiguration config, ExecutorService executorService) {
+    this.executorService = executorService;
+
+    this.logStream = config.awsCloudwatchlogStream().orElse("polaris-cloudwatch-default-stream");
+    this.logGroup = config.awsCloudwatchlogGroup().orElse("polaris-cloudwatch-default-group");
+    this.region = Region.of(config.awsCloudwatchRegion().orElse("us-east-1"));
+  }
+
+  @PostConstruct
+  void start() {
+    this.client = createCloudWatchClient();
+    ensureLogGroupAndStream();
+    backgroundTask = executorService.submit(this::processQueue);
+  }
+
+  protected CloudWatchLogsClient createCloudWatchClient() {
+    return CloudWatchLogsClient.builder().region(region).build();
+  }
+
+  private void processQueue() {
+    while (running || !queue.isEmpty()) {
+      drainQueue();
+    }
+  }
+
+  @VisibleForTesting
+  public void drainQueue() {
+    List<EventAndTimestamp> drainedEvents = new ArrayList<>();
+    List<InputLogEvent> transformedEvents = new ArrayList<>();
+    try {
+      EventAndTimestamp first = queue.poll(MAX_WAIT_MS, TimeUnit.MILLISECONDS);
+
+      if (first != null) {
+        drainedEvents.add(first);
+        queue.drainTo(drainedEvents, MAX_BATCH_SIZE - 1);
+      } else {
+        return;
+      }
+
+      drainedEvents.forEach(event -> transformedEvents.add(createLogEvent(event)));
+
+      sendToCloudWatch(transformedEvents);
+    } catch (Exception e) {
+      LOGGER.error("Error writing logs to CloudWatch: {}", e.getMessage());
+      LOGGER.error("Events not logged: {}", transformedEvents);
```

Comment:
This could be quite verbose, can we just print one event or some kind of summary?